LambdaEngine/Source/Rendering/PaintMaskRenderer.cpp

```diff
@@ -288,13 +342,28 @@ namespace LambdaEngine
 
 		CommandList* pCommandList = m_ppRenderCommandLists[modFrameIndex];
 
-		if (m_RenderTargets.IsEmpty())
+		if (m_RenderTargets.IsEmpty() || s_Collisions.empty())
 		{
 			return;
 		}
 
 		m_ppRenderCommandAllocators[modFrameIndex]->Reset();
 		pCommandList->Begin(nullptr);
+
+		// Transfer current collision data
+		{
+			TSharedRef<Buffer> unwrapDataCopyBuffer = m_UnwrapDataCopyBuffers[modFrameIndex];
+
+			byte* pUniformMapping	= reinterpret_cast<byte*>(unwrapDataCopyBuffer->Map());
+			UnwrapData data			= s_Collisions.front();
+			s_Collisions.pop_front();
```

Comment:
We render at tops one additional collision per frame?

Reply:
Yes, that is down to the current implementation of the unwrap stage that we have. Depending on how necessary it is it might have to change in the future, this wont however need to change the public API function so it should be fine.